base/src/Data/Macaw/Analysis/RegisterUse.hs

```diff
@@ -1053,9 +1053,9 @@ assignmentCache = lens assignDeps (\s v -> s { assignDeps = v })
 -- CallRegs
 
 -- | Identifies demand information about a particular call.
-data CallRegs (arch :: Type) = 
+data CallRegs (arch :: Type) (ids :: Type) =
   CallRegs { callRegsFnType :: !(ArchFunType arch)
-           , callArgRegs :: [Some (ArchReg arch)]
+           , callArgValues :: [Some (Value arch ids)]
```

Comment:
Yes, this allows the demand analaysis to know what values were used rather than just registers so that demand analysis can track which assignments were needed.  Reopt is the main client I know of for this code, and it isn't yet supporting stack allocated variables, but that's planned during the option.